### ISSUE TITLE
Unobserved Grpc.Core.RpcException is thrown when OLTP endpoint is unreachable

### ISSUE THREAD

**Issue by hankovich** [open-telemetry/opentelemetry-dotnet#3624]
List of all OpenTelemetry NuGet packages:

```xml
<PackageReference Include="OpenTelemetry.Exporter.Console" Version="1.3.0" />
<PackageReference Include="OpenTelemetry.Exporter.Jaeger" Version="1.3.0" />
<PackageReference Include="OpenTelemetry.Exporter.OpenTelemetryProtocol" Version="1.3.0" />
<PackageReference Include="OpenTelemetry.Extensions.Hosting" Version="1.0.0-rc9.5" />
<PackageReference Include="OpenTelemetry.Instrumentation.AspNetCore" Version="1.0.0-rc9.5" />
<PackageReference Include="OpenTelemetry.Instrumentation.ElasticsearchClient" Version="1.0.0-beta.3" />
<PackageReference Include="OpenTelemetry.Instrumentation.Http" Version="1.0.0-rc9.5" />
<PackageReference Include="OpenTelemetry.Instrumentation.StackExchangeRedis" Version="1.0.0-rc9.7" />
```

TFM: `net6.0`

## Symptom

I have a service that uses OLTP exporter. I noticed that an `UnobservedTaskException` is thrown when OLTP `Endpoint` is unreachable.

**What is the expected behavior?**

No exceptions are thrown.

**What is the actual behavior?**

`UnobservedTaskException` is thrown in the GC finalizer thread.

## Reproduce

```csharp
using System;
using System.Threading.Tasks;
using Microsoft.AspNetCore.Builder;
using Microsoft.AspNetCore.Http;
using Microsoft.Extensions.DependencyInjection;
using Newtonsoft.Json;
using OpenTelemetry.Resources;
using OpenTelemetry.Trace;

_ = ProcessGcAsync();

TaskScheduler.UnobservedTaskException += (_, eventArgs) =>
{
    Console.WriteLine(JsonConvert.SerializeObject(eventArgs));
};

var webApplicationBuilder = WebApplication.CreateBuilder();

webApplicationBuilder
    .Services.AddOpenTelemetryTracing(builder =>
    {
        builder.Configure((_, providerBuilder) =>
        {
            providerBuilder.AddOtlpExporter(options =>
            {
                options.Endpoint = new Uri("http://opentelemetry-collector.jaeger.svc:4317"); // any unknown host
            });
        });

        var resourceBuilder = ResourceBuilder
            .CreateDefault()
            .AddService("name", serviceVersion: "version");

        builder
            .AddSource("name")
            .SetResourceBuilder(resourceBuilder)
            .AddAspNetCoreInstrumentation()
            .AddHttpClientInstrumentation();
    });

var webApplication = webApplicationBuilder.Build();

webApplication.MapGet("/", () => Results.Ok("hello world!"));
webApplication.Run();

static async Task ProcessGcAsync()
{
    while (true)
    {
        GC.Collect(2, GCCollectionMode.Forced, true, true);
        GC.WaitForPendingFinalizers();

        await Task.Delay(100);
    }
}
```
## Additional Context

After some requests to `http://localhost:5000` the following message is written to the console:

```json
{
   "Observed":false,
   "Exception":{
      "ClassName":"System.AggregateException",
      "Message":"A Task's exception(s) were not observed either by Waiting on the Task or accessing its Exception property. As a result, the unobserved exception was rethrown by the finalizer thread.",
      "Data":null,
      "InnerException":{
         "Status":{
            "StatusCode":14,
            "Detail":"Error starting gRPC call. HttpRequestException: Этот хост неизвестен. (opentelemetry-collector.jaeger.svc:4317) SocketException: Этот хост неизвестен.",
            "DebugException":{
               "StatusCode":null,
               "Message":"Этот хост неизвестен. (opentelemetry-collector.jaeger.svc:4317)",
               "Data":{
                  
               },
               "InnerException":{
                  "ClassName":"System.Net.Sockets.SocketException",
                  "Message":"Этот хост неизвестен.",
                  "Data":null,
                  "InnerException":null,
                  "HelpURL":null,
                  "StackTraceString":"   at System.Net.Sockets.Socket.AwaitableSocketAsyncEventArgs.ThrowException(SocketError error, CancellationToken cancellationToken)\r\n   at System.Net.Sockets.Socket.AwaitableSocketAsyncEventArgs.System.Threading.Tasks.Sources.IValueTaskSource.GetResult(Int16 token)\r\n   at System.Net.Sockets.Socket.<ConnectAsync>g__WaitForConnectWithCancellation|277_0(AwaitableSocketAsyncEventArgs saea, ValueTask connectTask, CancellationToken cancellationToken)\r\n   at System.Net.Http.HttpConnectionPool.ConnectToTcpHostAsync(String host, Int32 port, HttpRequestMessage initialRequest, Boolean async, CancellationToken cancellationToken)",
                  "RemoteStackTraceString":null,
                  "RemoteStackIndex":0,
                  "ExceptionMethod":null,
                  "HResult":-2147467259,
                  "Source":"System.Net.Sockets",
                  "WatsonBuckets":null,
                  "NativeErrorCode":11001
               },
               "HelpLink":null,
               "Source":"System.Net.Http",
               "HResult":-2147467259,
               "StackTrace":"   at System.Net.Http.HttpConnectionPool.ConnectToTcpHostAsync(String host, Int32 port, HttpRequestMessage initialRequest, Boolean async, CancellationToken cancellationToken)\r\n   at System.Net.Http.HttpConnectionPool.ConnectAsync(HttpRequestMessage request, Boolean async, CancellationToken cancellationToken)\r\n   at System.Net.Http.HttpConnectionPool.AddHttp2ConnectionAsync(HttpRequestMessage request)\r\n   at System.Threading.Tasks.TaskCompletionSourceWithCancellation`1.WaitWithCancellationAsync(CancellationToken cancellationToken)\r\n   at System.Net.Http.HttpConnectionPool.GetHttp2ConnectionAsync(HttpRequestMessage request, Boolean async, CancellationToken cancellationToken)\r\n   at System.Net.Http.HttpConnectionPool.SendWithVersionDetectionAndRetryAsync(HttpRequestMessage request, Boolean async, Boolean doRequestAuth, CancellationToken cancellationToken)\r\n   at System.Net.Http.DiagnosticsHandler.SendAsyncCore(HttpRequestMessage request, Boolean async, CancellationToken cancellationToken)\r\n   at System.Net.Http.RedirectHandler.SendAsync(HttpRequestMessage request, Boolean async, CancellationToken cancellationToken)\r\n   at Grpc.Net.Client.Internal.GrpcCall`2.RunCall(HttpRequestMessage request, Nullable`1 timeout)"
            }
         },
         "StatusCode":14,
         "Trailers":[
            
         ],
         "Message":"Status(StatusCode=\"Unavailable\", Detail=\"Error starting gRPC call. HttpRequestException: Этот хост неизвестен. (opentelemetry-collector.jaeger.svc:4317) SocketException: Этот хост неизвестен.\", DebugException=\"System.Net.Http.HttpRequestException: Этот хост неизвестен. (opentelemetry-collector.jaeger.svc:4317)\r\n ---> System.Net.Sockets.SocketException (11001): Этот хост неизвестен.\r\n   at System.Net.Sockets.Socket.AwaitableSocketAsyncEventArgs.ThrowException(SocketError error, CancellationToken cancellationToken)\r\n   at System.Net.Sockets.Socket.AwaitableSocketAsyncEventArgs.System.Threading.Tasks.Sources.IValueTaskSource.GetResult(Int16 token)\r\n   at System.Net.Sockets.Socket.<ConnectAsync>g__WaitForConnectWithCancellation|277_0(AwaitableSocketAsyncEventArgs saea, ValueTask connectTask, CancellationToken cancellationToken)\r\n   at System.Net.Http.HttpConnectionPool.ConnectToTcpHostAsync(String host, Int32 port, HttpRequestMessage initialRequest, Boolean async, CancellationToken cancellationToken)\r\n   --- End of inner exception stack trace ---\r\n   at System.Net.Http.HttpConnectionPool.ConnectToTcpHostAsync(String host, Int32 port, HttpRequestMessage initialRequest, Boolean async, CancellationToken cancellationToken)\r\n   at System.Net.Http.HttpConnectionPool.ConnectAsync(HttpRequestMessage request, Boolean async, CancellationToken cancellationToken)\r\n   at System.Net.Http.HttpConnectionPool.AddHttp2ConnectionAsync(HttpRequestMessage request)\r\n   at System.Threading.Tasks.TaskCompletionSourceWithCancellation`1.WaitWithCancellationAsync(CancellationToken cancellationToken)\r\n   at System.Net.Http.HttpConnectionPool.GetHttp2ConnectionAsync(HttpRequestMessage request, Boolean async, CancellationToken cancellationToken)\r\n   at System.Net.Http.HttpConnectionPool.SendWithVersionDetectionAndRetryAsync(HttpRequestMessage request, Boolean async, Boolean doRequestAuth, CancellationToken cancellationToken)\r\n   at System.Net.Http.DiagnosticsHandler.SendAsyncCore(HttpRequestMessage request, Boolean async, CancellationToken cancellationToken)\r\n   at System.Net.Http.RedirectHandler.SendAsync(HttpRequestMessage request, Boolean async, CancellationToken cancellationToken)\r\n   at Grpc.Net.Client.Internal.GrpcCall`2.RunCall(HttpRequestMessage request, Nullable`1 timeout)\")",
         "Data":{
            
         },
         "InnerException":null,
         "HelpLink":null,
         "Source":"System.Private.CoreLib",
         "HResult":-2146233088,
         "StackTrace":"   at Grpc.Net.Client.Internal.HttpClientCallInvoker.BlockingUnaryCall[TRequest,TResponse](Method`2 method, String host, CallOptions options, TRequest request)\r\n   at Grpc.Core.Interceptors.InterceptingCallInvoker.<BlockingUnaryCall>b__3_0[TRequest,TResponse](TRequest req, ClientInterceptorContext`2 ctx)\r\n   at Grpc.Core.ClientBase.ClientBaseConfiguration.ClientBaseConfigurationInterceptor.BlockingUnaryCall[TRequest,TResponse](TRequest request, ClientInterceptorContext`2 context, BlockingUnaryCallContinuation`2 continuation)\r\n   at Grpc.Core.Interceptors.InterceptingCallInvoker.BlockingUnaryCall[TRequest,TResponse](Method`2 method, String host, CallOptions options, TRequest request)\r\n   at Opentelemetry.Proto.Collector.Trace.V1.TraceService.TraceServiceClient.Export(ExportTraceServiceRequest request, CallOptions options)\r\n   at Opentelemetry.Proto.Collector.Trace.V1.TraceService.TraceServiceClient.Export(ExportTraceServiceRequest request, Metadata headers, Nullable`1 deadline, CancellationToken cancellationToken)\r\n   at OpenTelemetry.Exporter.OpenTelemetryProtocol.Implementation.ExportClient.OtlpGrpcTraceExportClient.SendExportRequest(ExportTraceServiceRequest request, CancellationToken cancellationToken)"
      },
      "HelpURL":null,
      "StackTraceString":null,
      "RemoteStackTraceString":null,
      "RemoteStackIndex":0,
      "ExceptionMethod":null,
      "HResult":-2146233088,
      "Source":null,
      "WatsonBuckets":null,
      "InnerExceptions":[
         {
            "Status":{
               "StatusCode":14,
               "Detail":"Error starting gRPC call. HttpRequestException: Этот хост неизвестен. (opentelemetry-collector.jaeger.svc:4317) SocketException: Этот хост неизвестен.",
               "DebugException":{
                  "StatusCode":null,
                  "Message":"Этот хост неизвестен. (opentelemetry-collector.jaeger.svc:4317)",
                  "Data":{
                     
                  },
                  "InnerException":{
                     "ClassName":"System.Net.Sockets.SocketException",
                     "Message":"Этот хост неизвестен.",
                     "Data":null,
                     "InnerException":null,
                     "HelpURL":null,
                     "StackTraceString":"   at System.Net.Sockets.Socket.AwaitableSocketAsyncEventArgs.ThrowException(SocketError error, CancellationToken cancellationToken)\r\n   at System.Net.Sockets.Socket.AwaitableSocketAsyncEventArgs.System.Threading.Tasks.Sources.IValueTaskSource.GetResult(Int16 token)\r\n   at System.Net.Sockets.Socket.<ConnectAsync>g__WaitForConnectWithCancellation|277_0(AwaitableSocketAsyncEventArgs saea, ValueTask connectTask, CancellationToken cancellationToken)\r\n   at System.Net.Http.HttpConnectionPool.ConnectToTcpHostAsync(String host, Int32 port, HttpRequestMessage initialRequest, Boolean async, CancellationToken cancellationToken)",
                     "RemoteStackTraceString":null,
                     "RemoteStackIndex":0,
                     "ExceptionMethod":null,
                     "HResult":-2147467259,
                     "Source":"System.Net.Sockets",
                     "WatsonBuckets":null,
                     "NativeErrorCode":11001
                  },
                  "HelpLink":null,
                  "Source":"System.Net.Http",
                  "HResult":-2147467259,
                  "StackTrace":"   at System.Net.Http.HttpConnectionPool.ConnectToTcpHostAsync(String host, Int32 port, HttpRequestMessage initialRequest, Boolean async, CancellationToken cancellationToken)\r\n   at System.Net.Http.HttpConnectionPool.ConnectAsync(HttpRequestMessage request, Boolean async, CancellationToken cancellationToken)\r\n   at System.Net.Http.HttpConnectionPool.AddHttp2ConnectionAsync(HttpRequestMessage request)\r\n   at System.Threading.Tasks.TaskCompletionSourceWithCancellation`1.WaitWithCancellationAsync(CancellationToken cancellationToken)\r\n   at System.Net.Http.HttpConnectionPool.GetHttp2ConnectionAsync(HttpRequestMessage request, Boolean async, CancellationToken cancellationToken)\r\n   at System.Net.Http.HttpConnectionPool.SendWithVersionDetectionAndRetryAsync(HttpRequestMessage request, Boolean async, Boolean doRequestAuth, CancellationToken cancellationToken)\r\n   at System.Net.Http.DiagnosticsHandler.SendAsyncCore(HttpRequestMessage request, Boolean async, CancellationToken cancellationToken)\r\n   at System.Net.Http.RedirectHandler.SendAsync(HttpRequestMessage request, Boolean async, CancellationToken cancellationToken)\r\n   at Grpc.Net.Client.Internal.GrpcCall`2.RunCall(HttpRequestMessage request, Nullable`1 timeout)"
               }
            },
            "StatusCode":14,
            "Trailers":[
               
            ],
            "Message":"Status(StatusCode=\"Unavailable\", Detail=\"Error starting gRPC call. HttpRequestException: Этот хост неизвестен. (opentelemetry-collector.jaeger.svc:4317) SocketException: Этот хост неизвестен.\", DebugException=\"System.Net.Http.HttpRequestException: Этот хост неизвестен. (opentelemetry-collector.jaeger.svc:4317)\r\n ---> System.Net.Sockets.SocketException (11001): Этот хост неизвестен.\r\n   at System.Net.Sockets.Socket.AwaitableSocketAsyncEventArgs.ThrowException(SocketError error, CancellationToken cancellationToken)\r\n   at System.Net.Sockets.Socket.AwaitableSocketAsyncEventArgs.System.Threading.Tasks.Sources.IValueTaskSource.GetResult(Int16 token)\r\n   at System.Net.Sockets.Socket.<ConnectAsync>g__WaitForConnectWithCancellation|277_0(AwaitableSocketAsyncEventArgs saea, ValueTask connectTask, CancellationToken cancellationToken)\r\n   at System.Net.Http.HttpConnectionPool.ConnectToTcpHostAsync(String host, Int32 port, HttpRequestMessage initialRequest, Boolean async, CancellationToken cancellationToken)\r\n   --- End of inner exception stack trace ---\r\n   at System.Net.Http.HttpConnectionPool.ConnectToTcpHostAsync(String host, Int32 port, HttpRequestMessage initialRequest, Boolean async, CancellationToken cancellationToken)\r\n   at System.Net.Http.HttpConnectionPool.ConnectAsync(HttpRequestMessage request, Boolean async, CancellationToken cancellationToken)\r\n   at System.Net.Http.HttpConnectionPool.AddHttp2ConnectionAsync(HttpRequestMessage request)\r\n   at System.Threading.Tasks.TaskCompletionSourceWithCancellation`1.WaitWithCancellationAsync(CancellationToken cancellationToken)\r\n   at System.Net.Http.HttpConnectionPool.GetHttp2ConnectionAsync(HttpRequestMessage request, Boolean async, CancellationToken cancellationToken)\r\n   at System.Net.Http.HttpConnectionPool.SendWithVersionDetectionAndRetryAsync(HttpRequestMessage request, Boolean async, Boolean doRequestAuth, CancellationToken cancellationToken)\r\n   at System.Net.Http.DiagnosticsHandler.SendAsyncCore(HttpRequestMessage request, Boolean async, CancellationToken cancellationToken)\r\n   at System.Net.Http.RedirectHandler.SendAsync(HttpRequestMessage request, Boolean async, CancellationToken cancellationToken)\r\n   at Grpc.Net.Client.Internal.GrpcCall`2.RunCall(HttpRequestMessage request, Nullable`1 timeout)\")",
            "Data":{
               
            },
            "InnerException":null,
            "HelpLink":null,
            "Source":"System.Private.CoreLib",
            "HResult":-2146233088,
            "StackTrace":"   at Grpc.Net.Client.Internal.HttpClientCallInvoker.BlockingUnaryCall[TRequest,TResponse](Method`2 method, String host, CallOptions options, TRequest request)\r\n   at Grpc.Core.Interceptors.InterceptingCallInvoker.<BlockingUnaryCall>b__3_0[TRequest,TResponse](TRequest req, ClientInterceptorContext`2 ctx)\r\n   at Grpc.Core.ClientBase.ClientBaseConfiguration.ClientBaseConfigurationInterceptor.BlockingUnaryCall[TRequest,TResponse](TRequest request, ClientInterceptorContext`2 context, BlockingUnaryCallContinuation`2 continuation)\r\n   at Grpc.Core.Interceptors.InterceptingCallInvoker.BlockingUnaryCall[TRequest,TResponse](Method`2 method, String host, CallOptions options, TRequest request)\r\n   at Opentelemetry.Proto.Collector.Trace.V1.TraceService.TraceServiceClient.Export(ExportTraceServiceRequest request, CallOptions options)\r\n   at Opentelemetry.Proto.Collector.Trace.V1.TraceService.TraceServiceClient.Export(ExportTraceServiceRequest request, Metadata headers, Nullable`1 deadline, CancellationToken cancellationToken)\r\n   at OpenTelemetry.Exporter.OpenTelemetryProtocol.Implementation.ExportClient.OtlpGrpcTraceExportClient.SendExportRequest(ExportTraceServiceRequest request, CancellationToken cancellationToken)"
         }
      ]
   }
}
```

**Comment by cijothomas** [comment]
@vishweshbankwar Could you help triage this?

**Comment by vishweshbankwar** [comment]
looks like duplicate of https://github.com/open-telemetry/opentelemetry-dotnet/issues/4570. @hankovich @Handled1 Please try with the latest version of otlp package.

**Comment by Handled1** [comment]
> looks like duplicate of #4570. @hankovich @Handled1 Please try with the latest version of otlp package.

We've tested and can't see a reoccurrence.  Good from my side - really appreciate the follow up!

**Comment by vishweshbankwar** [comment]
Thanks for confirming @Handled1. Closing this issue
